homeassistant/components/modbus/climate.py

```diff
@@ -204,11 +220,41 @@ def __init__(
                     self._attr_fan_modes.append(fan_mode)
 
         else:
-            # No HVAC modes defined
+            # No FAN modes defined
             self._fan_mode_register = None
             self._attr_fan_mode = FAN_AUTO
             self._attr_fan_modes = [FAN_AUTO]
 
+        if CONF_SWING_MODE_REGISTER in config:
+            self._attr_supported_features = (
+                self._attr_supported_features | ClimateEntityFeature.SWING_MODE
+            )
+            mode_config = config[CONF_SWING_MODE_REGISTER]
+            self._swing_mode_register = mode_config[CONF_ADDRESS]
+            self._attr_swing_modes = cast(list[str], [])
+            self._attr_swing_mode = None
```

Comment:
STATE_UNKNOWN is used later, so either None or STATE_UNKNOWN is not correct.

Reply:
Yes STATE_UNKNOWN is used later, but it is reasonable to think that STATE_UNKNOWN is correct also at that time, I suppose, because at that time the component doesn't know its status yet...